src/stan/mcmc/chainset.hpp

```diff
@@ -0,0 +1,556 @@
+         << (param_names_.size() - 1);
+      throw std::invalid_argument(ss.str());
+    }
+    return param_names_[index];
+  }
+
+  /**
+   * Get column index for specified parameter name.
+   * Throws exception if name not found.
+   *
+   * @param name parameter name
+   * @return column index
+   */
+  int index(const std::string& name) const {
+    auto it = std::find(param_names_.begin(), param_names_.end(), name);
+    if (it == param_names_.end()) {
+      std::stringstream ss;
+      ss << "Unknown parameter name " << name;
+      throw std::invalid_argument(ss.str());
+    }
+    return std::distance(param_names_.begin(), it);
+  }
+
+  /**
+   * Assemble samples (draws) from specified column index across all chains
+   * into a matrix of samples X chain.
+   * Throws exception if column index is out of bounds.
+   *
+   * @param index column index
+   * @return matrix of draws across all chains
+   */
+  Eigen::MatrixXd samples(const int index) const {
+    Eigen::MatrixXd result(num_samples(), chains_.size());
+    if (index < 0 || index >= param_names_.size()) {
+      std::stringstream ss;
+      ss << "Bad index " << index << ", should be between 0 and "
+         << (param_names_.size() - 1);
+      throw std::invalid_argument(ss.str());
+    }
+    for (int i = 0; i < chains_.size(); ++i) {
+      result.col(i) = chains_[i].col(index);
+    }
+    return result;
+  }
+
+  /**
+   * Assemble samples (draws) from specified parameter name across all chains
+   * into a matrix of samples X chain.
+   * Throws exception if parameter name is not found.
+   *
+   * @param name parameter name
+   * @return matrix of draws across all chains
+   */
+  Eigen::MatrixXd samples(const std::string& name) const {
+    return samples(index(name));
+  }
+
+  /**
+   * Compute mean value for specified parameter across all chains.
+   *
+   * @param index parameter index
+   * @return mean parameter value
+   */
+  double mean(const int index) const { return samples(index).mean(); }
+
+  /**
+   * Compute mean value for specified parameter across all chains.
+   *
+   * @param name parameter name
+   * @return mean parameter value
+   */
+  double mean(const std::string& name) const { return mean(index(name)); }
+
+  /**
+   * Compute sample variance for specified parameter across all chains.
+   * 1 / (N - 1) * sum((theta_n - mean(theta))^2)
+   *
+   * @param index parameter index
+   * @return sample variance
+   */
+  double variance(const int index) const {
+    Eigen::MatrixXd draws = samples(index);
+    return (draws.array() - draws.mean()).square().sum() / (draws.size() - 1);
+  }
+
+  /**
+   * Compute sample variance for specified parameter across all chains.
+   * 1 / (N - 1) * sum((theta_n - mean(theta))^2)
+   *
+   * @param name parameter name
+   * @return sample variance
+   */
+  double variance(const std::string& name) const {
+    return variance(index(name));
+  }
```

Comment:
Where will these functions this used? I know the other class had these, but where they actually used? (I genuinely don't know)

Reply:
these are used by downstream analysis packages - afaik, only the CmdStan `stansummary` utility.

at the point where CmdStan became its own module, these stayed as part of Stan.  you'd have to ask the devs who were there - I don't know.

but I will edit the PR summary and description to better document all the changes in this PR.